src/lib.js

```diff
@@ -0,0 +1,331 @@
+export * from "./hamt/api.js"
+import * as Node from "./node.js"
+import { create as createBitmapIndexedNode } from "./node.js"
+import * as API from "./hamt/api.js"
+import * as Uint32Path from "./path/Uint32.js"
+import * as Uint8ArrayPath from "./path/Uint8Array.js"
+import { configure as configureBitField } from "./bitfield/lib.js"
+
+const NOT_FOUND = new RangeError("Not Found")
+
+/**
+ * A `bitWidth` determines the number of bits of the hash to use for index
+ * calculation at each level of the tree. In first layer will distribute
+ * children by the first `bitWidth` bits of the key hash. In the next layer
+ * next `bitWidth` number of bits of the key hash are used to determine
+ * placement of it's children and so on. Each node in the tree will hold
+ * `2 * bitWidth` number of elements.
+ *
+ * For example when we insert an entry named `result` it will generate a
+ * folowing hash (with murmur3 32) `442064690` which in binary would be
+ *
+ * ```
+ * 11010010110010101111100110010
+ * ```
+ *
+ * Which will correspond to a following key path
+ *
+ * ```
+ * 11010/01011/00101/01111/10011/0010
+ * ```
+ *
+ * Whis in decimals would be
+ *
+ * ```ts
+ * 26/11/5/15/19/2'
+ * ```
+ *
+ * If we then insert `fish` it would produce following path
+ *
+ * ```ts
+ * 26/3/4/18/28/19/1
+ * ```
+ * @template [V=unknown]
+ * @template {string} [K=string]
+ * @template {API.Config} [C=API.Config<API.Uint32>]
+ * @param {Partial<C>} [options]
+ * @returns {API.PersistentHashMap<V, K, C>}
+ */
+export const empty = options => {
+  const config = configure(options)
+  return new PersistentHashMap(0, createBitmapIndexedNode(config, null), config)
+}
+
+/**
+ * @template {API.Config} [C=API.Config<API.Uint32>]
+ * @param {Partial<C>} config
+ */
+const configure = ({
+  bitWidth = 5,
+  BitField = configureBitField({ bitWidth }),
+  Path = bitWidth === 5
+    ? Uint32Path.configure({ bitWidth })
+    : Uint8ArrayPath.configure({ bitWidth }),
+} = {}) => /** @type {C} */ ({ bitWidth, BitField, Path })
```

Comment:
Is there a good reason to have the `@type {C}` assertion inline here vs having an explicit `@returns` in the jsdoc?

Reply:
Yeah TS fails to infer it properly so this casts kind of like `<C>({...})` would do in pure TS. I'll add returns as well so it's less confusing